event-based-simulators/main/event_based_simulators.py

```diff
@@ -345,35 +277,42 @@ def __send_following_event(self, event_definition, timestamp = None, extra_param
                 followed_by_task = self.create_one_time_task(followed_by_definition)
                 followed_by_task.extra["timestamp"] = timestamp
                 followed_by_task.extra.update(extra_params)
-                self.tasks.append(followed_by_task)                
-                log.debug(f'{self.device_id} task({id(followed_by_task)}) added: {json.dumps(followed_by_definition)}, tasks: {len(self.tasks)}')      
+                self.tasks.append(followed_by_task)
+                log.debug(
+                    f'{self.device_id} task({id(followed_by_task)}) added: {json.dumps(followed_by_definition)}, tasks: {len(self.tasks)}')
             else:
-                  log.debug(f'{self.device_id} followedBy task missed. probability = {1 - followed_by_frequency / this_frequency} , def: {json.dumps(followed_by_definition)}')         
+                log.debug(
+                    f'{self.device_id} followedBy task missed. probability = {1 - followed_by_frequency / this_frequency} , def: {json.dumps(followed_by_definition)}')
 
-    def create_one_time_task(self, event_definition, start_in_seconds = 2, event_callback = None):
+    def create_one_time_task(self, event_definition, start_in_seconds=2, event_callback=None):
         callback = event_callback or MachineSimulator.event_mapping[event_definition["type"]]
         task = Task(start_in_seconds, lambda task: {self.__execute_callback_and_remove_task(callback, event_definition, task)})
-        return task        
-        
+        return task
+
     def __execute_callback_and_remove_task(self, callback, event_definition, task):
         callback(self, event_definition, task)
         if task in self.tasks:
             self.tasks.remove(task)
-            log.debug(f'{self.device_id} task({id(task)}) removed: {json.dumps(event_definition)}, tasks: {len(self.tasks)}')        
-    
+            log.debug(
+                f'{self.device_id} task({id(task)}) removed: {json.dumps(event_definition)}, tasks: {len(self.tasks)}')
+
     def tick(self):
         if not self.enabled: return
-        if not self.is_in_productionTime():
-            if not self.out_of_production_time_logged:
-                self.__log_not_in_shift()
-            self.out_of_production_time_logged = True
-            return
+        if self.current_work == "event":
```

Comment:
why is this only relevant for events?

Reply:
inside this condition, there are only functions related to event such as is_in_productionTime(), out_of_production_time_logged(), etc. 
measurement generation doesn't need these lines to work

Reply:
well, I think also measurements should only be sent during production time. @haraldmeyer @pejo-sag what do you think?

Reply:
I agree with Michael. Measurements are just a different way in which the devices send the data, so the same logic should apply to measurements and events.

e.g. you can have either "pieces produced" events from which you extract a piece count or you can have a measurement that represents a piece count. In both cases the assumption is that these should only happen during production time.